### ISSUE TITLE
False positive in `UnusedPrivateProperty` for private data class fields

### ISSUE THREAD

**Issue by lwasyl** [detekt/detekt#6142]
## Steps to Reproduce

```kotlin
data class Foo(
    private val foo: Int,
    private val bar: String,
)
```

will report `private val`s as unused. That's not the case — both are used in `equals` and `hashcode` implementations, and removing them changes semantics

## Your Environment

* Version of detekt used: 1.23.0
* Version of Gradle used (if applicable):
* Gradle scan link (add `--scan` option when running the gradle task):
* Operating System and version:
* Link to your project (if it's a public repository):

**Comment by cortinico** [comment]
Yup that's a valid report. We need `UnusedPrivateProperty` to exclude `data class`es
